test/metabase/upload_test.clj

```diff
@@ -358,58 +352,38 @@
       (let [header "a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,zz,yy,xx,ww,vv,uu,tt,ss,rr,qq,pp,oo,nn,mm,ll,kk,jj,ii,hh,gg,ff,ee,dd,cc,bb,aa"]
         (is (= (map keyword (str/split header #","))
                (keys
-                (:extant-columns
-                 (detect-schema-with-csv-rows
-                  [header
-                   "Luke,ah'm,yer,da,,,missing,columns,should,not,matter"])))))))
+                (detect-schema-with-csv-rows
+                 [header
+                  "Luke,ah'm,yer,da,,,missing,columns,should,not,matter"]))))))
     (testing "Empty contents (with header) are okay"
-      (is (=? (with-ai-id {:name     text-type
-                           :is_jedi_ text-type})
+      (is (=? {:name     text-type
+               :is_jedi_ text-type}
               (detect-schema-with-csv-rows
                ["Name, Is Jedi?"]))))
     (testing "Completely empty contents are okay"
-      (is (=? (with-ai-id {})
+      (is (=? {}
               (detect-schema-with-csv-rows
                [""]))))
     (testing "CSV missing data in the top row"
-      (is (=? (with-ai-id {:name       vchar-type
-                           :height     int-type
-                           :birth_year float-type})
+      (is (=? {:name       vchar-type
+               :height     int-type
+               :birth_year float-type}
               (detect-schema-with-csv-rows
                ["Name, Height, Birth Year"
                 ;; missing column
                 "Watto, 137"
                 "Luke Skywalker, 172, -19"
                 "Darth Vader, 202, -41.9"
                 ;; comma, but blank column
-                "Sebulba, 112,"]))))
-    (testing "Existing _mb_row_id column"
-      (is (=? {:extant-columns    {:ship       vchar-type
-                                   :name       vchar-type
-                                   :weapon     vchar-type}
-               :generated-columns {:_mb_row_id auto-pk-type}}
-              (detect-schema-with-csv-rows
-               ["_mb_row_id,ship,name,weapon"
```

Comment:
Do we have another test that covers how we treat having this row in the input for both values of the new feature flag?

Reply:
We have _almost_ equivalent tests, see `load-from-csv-auto-pk-column-test` and `load-from-csv-auto-pk-column-non-supporting-test`. I say almost equivalent because those tests don't explicitly check the `auto-pk-type`, but they do test that the type of the auto-pk column is an integer.